### PR TITLE
Add rector [batch]

### PR DESCRIPTION
**Pull request by xepozz**
Added rector/rector to CI and composer.json. Mass creation refers to https://github.com/yiisoft/yii-dev-tool/pull/232

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: tests/common/Dispatcher/CompositeDispatcherTest.php [hunk 1]
```diff
@@ -60,10 +60,7 @@ public function testPropagationStops(): void
         $this->assertSame($stoppableEvent, $result);
     }
 
-    /**
-     * @return \PHPUnit\Framework\MockObject\MockObject|\Psr\EventDispatcher\EventDispatcherInterface
-     */
-    private function createTransparentDispatcher()
+    private function createTransparentDispatcher(): \PHPUnit\Framework\MockObject\MockObject|EventDispatcherInterface
```
Comment (arogachev) [inline tests/common/Dispatcher/CompositeDispatcherTest.php:63]: Let's additionally add it to use statement?

File: .github/workflows/rector.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+name: rector
+
+jobs:
+  rector:
+    uses: yiisoft/actions/.github/workflows/rector.yml@master
+    with:
+      os: >-
+        ['ubuntu-latest']
+      php: >-
+        ['8.0']
```
Comment (arogachev) [inline .github/workflows/rector.yml:21]: Maybe include 8.1 here too?

**Review by arogachev — CHANGES_REQUESTED** [review]
:+1:

File: .github/workflows/rector.yml (hunk 2 above)
Comment (arogachev) [reply .github/workflows/rector.yml:21]: Let's handle it afterwards.

**Review by arogachev — APPROVED** [review]

**Pull request merged**